### PR TITLE
Add CNPs for postgres instances and join/init operations

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/giantswarm/issues/30093

This PR:

- adds Cilium Network Policies for CNPG database, init, and join pods

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/cloudnative-pg/templates/cilium-network-policy.yaml [hunk 1]
```diff
@@ -41,4 +42,74 @@ spec:
...
+        - ports:
+            - port: "8000"
+        - ports:
+            - port: "5432"
+    # Allow Prometheus scraping.
+    - fromEntities:
+        - cluster
+      toPorts:
+        - ports:
+            - port: "9187"
```
Comment (fhielpos) [inline helm/cloudnative-pg/templates/cilium-network-policy.yaml:74]: Why does it use such a weird port for scraping? 💀

File: helm/cloudnative-pg/templates/cilium-network-policy.yaml [hunk 2]
```diff
@@ -41,4 +42,74 @@ spec:
...
+  egress:
+    - toEntities:
+        - kube-apiserver
```
Comment (fhielpos) [inline helm/cloudnative-pg/templates/cilium-network-policy.yaml:77]: Why does this pod need api server? Is this one the actual operator?

**Review by fhielpos — APPROVED** [review]
Looks good to me, just a couple questions but nothing crazy.

File: helm/cloudnative-pg/templates/cilium-network-policy.yaml (hunk 1 above)
Comment (stone-z) [reply helm/cloudnative-pg/templates/cilium-network-policy.yaml:74]: No idea :man_shrugging:

File: helm/cloudnative-pg/templates/cilium-network-policy.yaml (hunk 2 above)
Comment (stone-z) [reply helm/cloudnative-pg/templates/cilium-network-policy.yaml:77]: There is a management process that sits next to postgres and updates cluster status in the API

**Pull request merged**